### ISSUE TITLE
Arbled characters when using snapshot_download and hf_hub_download to download files with cn/jp/kr names

### ISSUE THREAD

**Issue by narugo1992** [huggingface/huggingface_hub#1942]
### Describe the bug

When use `snapshot_download` or `hf_hub_download` to download files with chinese (both simpifiled and traditional), japanese or korean names, the tqdm will show arbled characters.

Run the following code

```python
from huggingface_hub import HfApi

hf_client = HfApi()
hf_client.snapshot_download(
    repo_id='narugo/test_rp',
    repo_type='dataset',
    local_dir='test_langs',
    local_dir_use_symlinks=False,
)

print('这是简体中文，输出是正常的')
print('This is simplified chinese, nothing wrong with the environment')
```

The result is

![image](https://github.com/huggingface/huggingface_hub/assets/117186571/7caaba21-8775-482a-ab2b-36e55f8e9718)

The chinese text in `print` statement will be printed normally, while text in `snapshot_download` will not.

The multi-lang repository for test we are using: https://huggingface.co/datasets/narugo/test_rp

### Reproduction

from huggingface_hub import HfApi

hf_client = HfApi()
hf_client.snapshot_download(
    repo_id='narugo/test_rp',
    repo_type='dataset',
    local_dir='test_langs',
    local_dir_use_symlinks=False,
)

print('这是简体中文，输出是正常的')
print('This is simplified chinese, nothing wrong with the environment')

### System info

```shell
Copy-and-paste the text below in your GitHub issue.

- huggingface_hub version: 0.20.1
- Platform: Linux-5.10.198-187.748.amzn2.x86_64-x86_64-with-glibc2.2.5
- Python version: 3.8.1
- Running in iPython ?: No
- Running in notebook ?: No
- Running in Google Colab ?: No
- Token path ?: /home/user/.cache/huggingface/token
- Has saved token ?: True
- Who am I ?: narugo
- Configured git credential helpers: 
- FastAI: N/A
- Tensorflow: N/A
- Torch: 2.1.0
- Jinja2: 3.1.2
- Graphviz: N/A
- Pydot: N/A
- Pillow: 9.5.0
- hf_transfer: N/A
- gradio: N/A
- tensorboard: N/A
- numpy: 1.24.4
- pydantic: N/A
- aiohttp: 3.8.6
- ENDPOINT: https://huggingface.co
- HF_HUB_CACHE: /home/user/.cache/huggingface/hub
- HF_ASSETS_CACHE: /home/user/.cache/huggingface/assets
- HF_TOKEN_PATH: /home/user/.cache/huggingface/token
- HF_HUB_OFFLINE: False
- HF_HUB_DISABLE_TELEMETRY: False
- HF_HUB_DISABLE_PROGRESS_BARS: None
- HF_HUB_DISABLE_SYMLINKS_WARNING: False
- HF_HUB_DISABLE_EXPERIMENTAL_WARNING: False
- HF_HUB_DISABLE_IMPLICIT_TOKEN: False
- HF_HUB_ENABLE_HF_TRANSFER: False
- HF_HUB_ETAG_TIMEOUT: 10
- HF_HUB_DOWNLOAD_TIMEOUT: 10
```

**Comment by Wauplin** [comment]
Hi @narugo1992, thanks for reporting. I've investigated a bit but not sure we can do much about it. The problem comes from the header returned by cloudfront in which the filename is encoded. Here is a standalone script to reproduce it:

```py
import requests
from huggingface_hub import hf_hub_url

url = hf_hub_url(repo_id="narugo/test_rp", filename="日本語/でこんにちは.jpg", repo_type="dataset")
response = requests.head(url=url, allow_redirects=True) # 1 redirect to cloudfront CDN
print(response.headers.get("content-disposition"))
```

which outputs
```
inline; filename*=UTF-8''%E3%81%A7%E3%81%93%E3%82%93%E3%81%AB%E3%81%A1%E3%81%AF.jpg; filename="ã§ããã«ã¡ã¯.jpg";
```

Filename is "badly" decoded by `requests` and since we're using this value in the tqdm line, I don't think we can do much :confused:

**Comment by narugo1992** [comment]
FIY, for the url of huggingface site resources, maybe extracting the file name from the url and using it for the desc parameter of tqdm is a solution. 

e.g. https://huggingface.co/datasets/narugo/test_rp/resolve/main/%E6%97%A5%E6%9C%AC%E8%AA%9E/%E3%81%A7%E3%81%93 %E3%82%93%E3%81%AB%E3%81%A1%E3%81%AF.jpg, just extract the file name through urlsplit, and obtain the decoded file name through unquote.

**Comment by Wauplin** [comment]
Thanks for your input @narugo1992. I just opened a PR to fix this (see https://github.com/huggingface/huggingface_hub/pull/1954). I went for the simplest solution which is to provide the filename explicitly -no encoding/decoding involved-. Relying on the URL directly was not possible as it wouldn't work for files hosted on a CDN (that's why we have read the `content-disposition` header which is error-prone).
